orca-kayenta/src/main/kotlin/com/netflix/spinnaker/orca/kayenta/pipeline/KayentaCanaryStage.kt

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright 2018 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.kayenta.pipeline
+
+import com.fasterxml.jackson.databind.SerializationFeature.WRITE_DATES_AS_TIMESTAMPS
+import com.fasterxml.jackson.module.kotlin.convertValue
+import com.netflix.spinnaker.orca.ext.mapTo
+import com.netflix.spinnaker.orca.ext.withTask
+import com.netflix.spinnaker.orca.jackson.OrcaObjectMapper
+import com.netflix.spinnaker.orca.kayenta.CanaryScope
+import com.netflix.spinnaker.orca.kayenta.CanaryScopes
+import com.netflix.spinnaker.orca.kayenta.model.KayentaCanaryContext
+import com.netflix.spinnaker.orca.kayenta.model.RunCanaryContext
+import com.netflix.spinnaker.orca.kayenta.tasks.AggregateCanaryResultsTask
+import com.netflix.spinnaker.orca.pipeline.StageDefinitionBuilder
+import com.netflix.spinnaker.orca.pipeline.StageDefinitionBuilder.newStage
+import com.netflix.spinnaker.orca.pipeline.TaskNode.Builder
+import com.netflix.spinnaker.orca.pipeline.WaitStage
+import com.netflix.spinnaker.orca.pipeline.model.Stage
+import com.netflix.spinnaker.orca.pipeline.model.SyntheticStageOwner.STAGE_BEFORE
+import org.springframework.stereotype.Component
+import java.time.Clock
+import java.time.Duration
+import java.time.Duration.ZERO
+import java.time.Instant
+import java.time.Instant.now
+import java.time.temporal.ChronoUnit.MINUTES
+import java.util.*
+import java.util.Collections.singletonMap
+
+@Component
+class KayentaCanaryStage(
+  private val clock: Clock,
+  private val waitStage: WaitStage
+) : StageDefinitionBuilder {
+
+  private val mapper = OrcaObjectMapper
+    .newInstance()
+    .disable(WRITE_DATES_AS_TIMESTAMPS) // we want Instant serialized as ISO string
```

Comment:
I wonder why this isn't serializing dates correctly

Reply:
Eh. It's because that mapper is only used by `mapTo` not when actually serializing the entire pipeline / stage